File: 1-CLEAN_CODE/SOLID/OpenClose/.NET/dont/Person.cs
using System;
using CraftsmanshipPath.Solid.SingleResponsability;

namespace CraftsmanshipPath.Solid.SingleResponsability
{
    public class Person
    {
        public Person(string firstName, string lastName)
        {
            FirstName = firstName;
            LastName = lastName;
        }

        public  string FirstName { get; set; } // DO NOT set the property public
        public string LastName { get; set; } // DO NOT set the property public
    }

    public class Employee : Person
    {
        public Employee(string firstName, string lastName, string matricule): base(firstName, lastName)
        {
            FirstName = firstName;
            LastName = lastName;
            Matricule = matricule;
        }
        public string Matricule { get; }
    }
}

File: 1-CLEAN_CODE/SOLID/OpenClose/TS/dont/person.ts
namespace CraftsmanshipPath.Solid.OpenClose {
  class Person {
    public readonly firstName;
    public readonly lastName;
    public readonly matricule; // new feature: adding matrivule to person
    
    constructor(firstName, lastName, matricule) { // new feature: adding matrivule to person
      this.firstName = firstName;
      this.lastName = lastName;
      this.matricule = matricule
    }

  }  
}
